unit2-java/lesson08.rst

```diff
@@ -4,27 +4,20 @@ Unit 2 (Java) - Lesson 8 Learning Objectives
 Goals
 -----
 
+- Understand the basic architecture of an MVC web application
+- Build a Spring Boot application that can handle various types of requests
+
 Objectives
 ----------
 
-- Describe what an IDE like IntelliJ offers programmers that text
-   editors do not
-- Run your program in an IDE
-- Use keyboard shortcuts in IntelliJ
-- Easily refactor and rename in IntelliJ
-- Diagnose logical bugs using IntelliJ’s debugging tools
-- Step through code in debugging mode
-- Create a repo and connect it to Git in IntelliJ
-- Create a ``.gitignore`` file
-- Connect to a remote, ``commit``, and ``push`` using Git within IntelliJ
-- Create and ``fetch`` from an upstream branch
+- Review HTTP basics 
 - Describe the MVC pattern
 - Create simple Spring Boot projects using Gradle
+- Describe the basic functionality the Gradle provides: dependency management, build tasks
 - Run a Spring Boot web app in IntelliJ
 - Understand the role of controllers in a Java web app
 - Create and modify controllers to handle different routes and request types
-- Use the annotations ``@RequestMapping``, ``@Controller``, and ``@ResponseBody``
-- Get data from users via ``GET``, ``POST``, and URL segments
-- Manipulate and return user-data in your controller
-- Use the classes ``Model`` and ``HttpServletRequest``
-- Redirect users to a desired route
+- Use the annotations ``GetMapping``, ``PostMapping``, ``@RequestMapping``, ``@Controller``, and ``@ResponseBody``
+- Get data from users via ``GET``, ``POST``, and URL segments (using ``@PathVariable``)
+- Return plain-text responses from a controller 
+- Use ``HttpServletRequest`` to access query parameters
```

Comment:
Will we also be adding a redirect?  The old videos ended with a re-routing to a form with ``public String goodbye(){ return "redirect:/";}``.

Also, what is the difference between ``@GetMapping`` and ``@RequestMapping``?  You only used the latter in the old videos.

Reply:
I took out redirects since I don't know if there's a direct use-case in following lessons. We can always introduce this if/when it is needed later, but I'm not sure it will come up.